Remove TURN_NO_TLS and TLS_SUPPORTED from code

eakraly opened the pull request:
`TURN_NO_TLS` can only be set externally to `configure` (which probably never happens or even possible with code as all openssl includes are mandatory)
As such, `TLS_SUPPORTED` is always set to true.

Removing both from code makes it a little cleaner

ggarber reviewed (commented):
Can you clarify the impact?

With this you won't be able to disable TLS support at compile time, right?  That's the non backwards compatible change added and we assume that nobody is using it?

Doesn't it make it inconsistent with TURN_NO_DTLS for example ?

I'm a fan any PR that removes things but I would like to understand it better.  In this case it is hard (for me) to understand if this will have unintended impact for somebody.  Thx!

eakraly commented on src/apps/common/apputils.h:
```diff
@@ -65,35 +65,22 @@ extern int IS_TURN_SERVER;
-	#define TLSv1_1_SUPPORTED 0
-	#define TLSv1_2_SUPPORTED 0
-
+#if defined(SSL_OP_NO_TLSv1_1)
```
Good question and great suggestion.
I do not know the history of this but yes - redefining macros is questionable

eakraly commented:
>Can you clarify the impact?

>With this you won't be able to disable TLS support at compile time, right? That's the non backwards compatible change added and we assume that nobody is using it?

Correct, cannot disable TLS during compile time. Can disable TLS during runtime.

> Doesn't it make it inconsistent with TURN_NO_DTLS for example ?

Correct, fixed. TURN_NO_DTLS is also removed and now detected based on OpenSSL version. But no ability to compile turnserver without openssl

> I'm a fan any PR that removes things but I would like to understand it better. In this case it is hard (for me) to understand if this will have unintended impact for somebody. Thx!

As you properly mentioned this change now REQUIRES openssl. To be fair, it is not clear if before this change it is at all possible to build coturn without openssl. And if there is openssl then there is SSL/TLS, if there is TLS then TURN_NO_TLS is redundant etc.

If for any reason, someone needs coturn WITHOUT openssl (and by extent no auth operation whatsoever, or HTTPS - including DB, prometheus, redis, web admin interface) then the only remedy is to use 4.6.0.

eakraly commented:
Thanks @ggarber 
Good feedback
Let me rework the suggestion to keep the ability to build without TLS/openssl while simplifying the defines
I'll close this PR
